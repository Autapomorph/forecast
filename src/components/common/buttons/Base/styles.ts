import styled from 'styled-components/macro';

export const IconButton = styled.button.attrs({
  type: 'button',
})`
  padding: 5px;
  border: none;
  background: none;
  color: inherit;
  cursor: pointer;

  :hover {
    color: var(--white-dark);
  }
`;
