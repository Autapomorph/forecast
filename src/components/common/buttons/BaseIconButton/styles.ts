import styled from 'styled-components';

// eslint-disable-next-line import/prefer-default-export
export const StyledIconButton = styled.span`
  cursor: pointer;

  :hover {
    color: var(--light-gray);
  }
`;
