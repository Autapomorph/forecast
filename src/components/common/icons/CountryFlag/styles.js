import styled, { css } from 'styled-components';

// eslint-disable-next-line import/prefer-default-export
export const StyledCountryFlag = styled.span`
  display: inline-block;
  position: relative;
  width: 1.33333333em;
  background-image: ${({ country }) => `url(/images/countries/flags/4x3/${country}.svg)`};
  background-position: 50%;
  background-size: contain;
  background-repeat: no-repeat;

  ::before {
    content: '\00a0';
  }

  ${
    /* sc-dec */ ({ background }) =>
      background &&
      css`
        background-position: 50%;
        background-size: contain;
        background-repeat: no-repeat;
      `
  }

  ${
    /* sc-dec */ ({ squared }) =>
      squared &&
      css`
        width: 1em;
        background-image: ${({ country }) => `url(/images/countries/flags/1x1/${country}.svg)`};
      `
  }

  ${
    /* sc-dec */ ({ size }) =>
      size &&
      css`
        font-size: ${size};
      `
  }
`;
