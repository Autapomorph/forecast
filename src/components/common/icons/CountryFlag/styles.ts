import styled from 'styled-components/macro';

type Props = {
  country: string;
};

export const CountryFlag = styled.span`
  display: inline-block;
  width: 1em;
  height: 1em;
  border-radius: 50%;
  vertical-align: middle;
  background-image: ${({ country }: Props) => `url(/images/countries/flags/${country}.svg) `};
  background-position: center center;
  background-repeat: no-repeat;
  background-size: cover;
`;
