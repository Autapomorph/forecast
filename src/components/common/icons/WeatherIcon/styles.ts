import styled from 'styled-components/macro';

import 'weathericons/css/weather-icons.css';
import 'weathericons/css/weather-icons-wind.css';

export const WeatherIcon = styled.span`
  vertical-align: middle;
`;
