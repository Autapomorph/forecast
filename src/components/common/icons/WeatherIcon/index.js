import React from 'react';

import { StyledWeatherIcon } from './styles';

const WeatherIcon = ({
  tag = 'i',
  icon,
  wind = false,
  fixedWidth,
  flip,
  rotate,
  size,
  className,
}) => {
  const classNames = ['wi'];
  if (wind || /^wi-/i.test(icon)) {
    classNames.push(icon);
  } else {
    classNames.push(`wi-${icon}`);
  }

  if (wind) classNames.push(`wi-wind`);
  if (size) classNames.push(`wi-size-${size}`);
  if (fixedWidth) classNames.push('wi-fw');
  if (flip) classNames.push(`wi-flip-${flip}`);
  if (rotate) classNames.push(`wi-rotate-${rotate}`);
  if (className) classNames.push(className);

  let fontSize;
  switch (size) {
    case 'lg':
      fontSize = '1.33333333em';
      break;
    case '2x':
      fontSize = '2em';
      break;
    case '3x':
      fontSize = '3em';
      break;
    case '4x':
      fontSize = '4em';
      break;
    case '5x':
      fontSize = '5em';
      break;
    default:
      fontSize = 'normal';
      break;
  }

  return <StyledWeatherIcon as={tag} className={classNames.join(' ')} size={fontSize} />;
};

export default WeatherIcon;
