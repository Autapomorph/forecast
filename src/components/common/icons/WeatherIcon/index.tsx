import React from 'react';

import * as S from './styles';

type Props = {
  icon: string;
  wind?: boolean;
  fixedWidth?: boolean;
  flip?: 'horizontal' | 'vertical' | false;
  rotate?: number | false;
  className?: string;
  style?: React.CSSProperties;
};

const WeatherIcon = ({
  icon,
  wind = false,
  fixedWidth = false,
  flip = false,
  rotate = false,
  className = '',
  style,
}: Props): React.ReactElement => {
  const classNames = ['wi'];
  if (wind || /^wi-/i.test(icon)) {
    classNames.push(icon);
  } else {
    classNames.push(`wi-${icon}`);
  }

  if (wind) classNames.push(`wi-wind`);
  if (fixedWidth) classNames.push('wi-fw');
  if (flip) classNames.push(`wi-flip-${flip}`);
  if (rotate) classNames.push(`wi-rotate-${rotate}`);
  if (className) classNames.push(className);

  return <S.WeatherIcon className={classNames.join(' ')} style={style} />;
};

export default WeatherIcon;
