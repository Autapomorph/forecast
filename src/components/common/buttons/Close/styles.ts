import styled from 'styled-components/macro';

import { ThemeProp } from 'models';
import BaseIconButton from 'components/common/buttons/Base';

export const CloseButton = styled(BaseIconButton)`
  padding: 5px 10px;
  color: ${({ theme }: ThemeProp) => theme.baseTextColor};
  font-size: 1.2rem;

  :hover {
    color: var(--gray-light-lighter);
  }
`;
