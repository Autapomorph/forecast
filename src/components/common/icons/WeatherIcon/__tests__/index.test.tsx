import React from 'react';
import { shallow } from 'enzyme';

import WeatherIcon from '..';

import * as S from '../styles';

describe('WeatherIcon tests', () => {
  it('renders properly', () => {
    const wrapper = shallow(<WeatherIcon icon="north" />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render with icon class', () => {
    const wrapper = shallow(<WeatherIcon icon="north" />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(S.WeatherIcon).hasClass('wi-north')).toBeTruthy();
  });

  it('should render with wind class', () => {
    const wrapper = shallow(<WeatherIcon icon="north" wind />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(S.WeatherIcon).hasClass('wi-wind')).toBeTruthy();
  });

  it('should render with fixedWidth class', () => {
    const wrapper = shallow(<WeatherIcon icon="north" fixedWidth />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(S.WeatherIcon).hasClass('wi-fw')).toBeTruthy();
  });

  it('should render with flip class', () => {
    const wrapper = shallow(<WeatherIcon icon="north" flip="horizontal" />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(S.WeatherIcon).hasClass('wi-flip-horizontal')).toBeTruthy();
  });

  it('should render with rotate class', () => {
    const wrapper = shallow(<WeatherIcon icon="north" rotate={90} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(S.WeatherIcon).hasClass('wi-rotate-90')).toBeTruthy();
  });

  it('should preserve passed class', () => {
    const wrapper = shallow(<WeatherIcon icon="north" className="class" />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(S.WeatherIcon).hasClass('class')).toBeTruthy();
  });
});
