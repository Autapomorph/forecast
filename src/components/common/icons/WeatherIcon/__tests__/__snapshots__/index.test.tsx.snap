// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherIcon tests renders properly 1`] = `
<styles__WeatherIcon
  className="wi wi-north"
/>
`;

exports[`WeatherIcon tests should preserve passed class 1`] = `
<styles__WeatherIcon
  className="wi wi-north class"
/>
`;

exports[`WeatherIcon tests should render with fixedWidth class 1`] = `
<styles__WeatherIcon
  className="wi wi-north wi-fw"
/>
`;

exports[`WeatherIcon tests should render with flip class 1`] = `
<styles__WeatherIcon
  className="wi wi-north wi-flip-horizontal"
/>
`;

exports[`WeatherIcon tests should render with icon class 1`] = `
<styles__WeatherIcon
  className="wi wi-north"
/>
`;

exports[`WeatherIcon tests should render with rotate class 1`] = `
<styles__WeatherIcon
  className="wi wi-north wi-rotate-90"
/>
`;

exports[`WeatherIcon tests should render with wind class 1`] = `
<styles__WeatherIcon
  className="wi north wi-wind"
/>
`;
